editor/editor_vcs_interface.cpp

```diff
@@ -54,58 +62,107 @@ void EditorVCSInterface::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("shut_down"), &EditorVCSInterface::shut_down);
 	ClassDB::bind_method(D_METHOD("get_project_name"), &EditorVCSInterface::get_project_name);
 	ClassDB::bind_method(D_METHOD("get_vcs_name"), &EditorVCSInterface::get_vcs_name);
+	ClassDB::bind_method(D_METHOD("get_previous_commits"), &EditorVCSInterface::get_previous_commits);
+	ClassDB::bind_method(D_METHOD("get_branch_list"), &EditorVCSInterface::get_branch_list);
+	ClassDB::bind_method(D_METHOD("checkout_branch", "branch"), &EditorVCSInterface::checkout_branch);
+	ClassDB::bind_method(D_METHOD("push"), &EditorVCSInterface::push);
+	ClassDB::bind_method(D_METHOD("pull"), &EditorVCSInterface::pull);
+	ClassDB::bind_method(D_METHOD("fetch"), &EditorVCSInterface::fetch);
+	ClassDB::bind_method(D_METHOD("set_up_credentials"), &EditorVCSInterface::set_up_credentials);
 }
 
 bool EditorVCSInterface::_initialize(String p_project_root_path) {
 
-	WARN_PRINT("Selected VCS addon does not implement an initialization function. This warning will be suppressed.")
+	WARN_PRINT("Selected VCS addon does not implement  initialization function. This warning will be suppressed.");
 	return true;
 }
 
 bool EditorVCSInterface::_is_vcs_initialized() {
-
+	WARN_PRINT("Selected VCS addon does not implement \"" + String(__FUNCTION__) + "\"function. This warning will be suppressed.")
 	return false;
 }
 
 Dictionary EditorVCSInterface::_get_modified_files_data() {
-
+	WARN_PRINT("Selected VCS addon does not implement  \"" + String(__FUNCTION__) + "\" function. This warning will be suppressed.")
 	return Dictionary();
 }
 
 void EditorVCSInterface::_stage_file(String p_file_path) {
+	WARN_PRINT("Selected VCS addon does not implement  \"" + String(__FUNCTION__) + "\" function. This warning will be suppressed.")
 }
 
 void EditorVCSInterface::_unstage_file(String p_file_path) {
+	WARN_PRINT("Selected VCS addon does not implement  \"" + String(__FUNCTION__) + "\" function. This warning will be suppressed.")
 }
 
 void EditorVCSInterface::_commit(String p_msg) {
+	WARN_PRINT("Selected VCS addon does not implement  \"" + String(__FUNCTION__) + "\" function. This warning will be suppressed.")
+}
+
+void EditorVCSInterface::_discard_file(String p_file_path){
+	WARN_PRINT("Selected VCS addon does not implement  \"" + String(__FUNCTION__) + "\" function. This warning will be suppressed.")
 }
 
 Array EditorVCSInterface::_get_file_diff(String p_file_path) {
+	WARN_PRINT("Selected VCS addon does not implement  \"" + String(__FUNCTION__) + "\" function. This warning will be suppressed.")
+	return Array();
+}
 
+Array EditorVCSInterface::_get_branch_list() {
+	WARN_PRINT("Selected VCS addon does not implement  \"" + String(__FUNCTION__) + "\" function. This warning will be suppressed.")
 	return Array();
 }
 
-bool EditorVCSInterface::_shut_down() {
+bool EditorVCSInterface::_checkout_branch(String branch) {
```

Comment:
Godot follows generally a convention of `p_` and `r_` (for references) prefix, as you already have been using. There are a few places where this isn't used in this PR. You can find those easily by going through the diff.

This is not high priority but yes, this will be required to get this PR merged in finally